pkg/recipes/terraform/execute.go

```diff
@@ -88,10 +91,12 @@ func (e *executor) Deploy(ctx context.Context, options Options) (*tfjson.State,
 		return nil, err
 	}
 
-	// Set environment variables for the Terraform process.
-	err = e.setEnvironmentVariables(ctx, tf, options.EnvConfig)
-	if err != nil {
-		return nil, err
+	if options.EnvConfig != nil {
+		// Set environment variables for the Terraform process.
```

Comment:
I don't think we need the comment here. The function name is clear enough.

Reply:
keeping the comment for now, in line with rest of the function